contracts/mock/SimpleOracle.sol

```diff
@@ -16,11 +16,18 @@ contract SimpleOracle is IOracle {
         return true;
     }
 
-    function changePrice(address token, uint newPrice) external {
+    function changePrice(address token, int newPrice) external {
         prices[token] = newPrice;
     }
 
-    function getLatestAnswer(address token) external returns(uint256) {
+    function getLatestAnswer(address token) external returns(int256) {
+        // mimic eip4626
+        (bool success, bytes memory result) = token.call(abi.encodeWithSignature("asset()"));
+        if(success && result.length > 0) {
+            // get the underlying token value (if ERC4626)
+            // NB: Share token to underlying ratio might not be 1:1
+            token = abi.decode(result, (address));
```

Comment:
For the sake of simplicity I had it return the price of one unit of the underlying and then handled the conversion rate within the Escrow contract. The reason being that this seems like an opinionated change and we have to scale each token by decimals already in other contracts. 

Also of note: the 1e8 figure is because it is quoted in USD, every token (regardless of their decimals) will be quoted back in 1e8 since 1e18 represents USD. This assumption might not be true for other pairs e.g. WBTC/ETH but it is the case for USD. As this is the case, the chainlink oracle will return the underlying token price in 1e8 and we will have to scale it by the token decimals elsewhere, unless we want to move this logic into the oracle and remove the scaling done in other contracts like the Escrow.